code/modules/paperwork/stamps.dm

```diff
@@ -12,6 +12,78 @@
 	item_color = "cargo"
 	pressure_resistance = 2
 	attack_verb = list("stamped")
+	var/stamp_color = "#264715"		//used for stamp marks, based on the item sprites.. not the on paper images(those are very bland)
+	var/static/list/stamp_examine_list = list(
+		"stamp-qm" = "Quartermaster approved",
+		"stamp-law" = "Justice Department approved",
+		"stamp-cap" = "Captain approved",
+		"stamp-hop" = "Head of Personnel approved",
+		"stamp-hos" = "Head of Security approved",
+		"stamp-ce" = "Chief Engineer approved",
+		"stamp-rd" = "Research Director approved",
+		"stamp-cmo" = "Chief Medical Officer approved",
+		"stamp-ok" = "GRANTED",
+		"stamp-deny" = "DENIED",
+		"stamp-clown" = "HONK",
+		"stamp-rep" = "Nanotrasen Representative approved",
+		"stamp-magistrate" = "Magistrate approved",
+		"stamp-cent" = "Central Command approved",
+		"stamp-syndicate" = "Syndicate approved")
+
+/obj/item/stamp/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	if(!ishuman(target) || !ishuman(user) || !proximity)
+		return
+
+	var/mob/living/carbon/human/H = target
+	var/list/paramslist = list()
+	var/attackedSide				//the stamp mark will appear on this side, NORTH = forwards, SOUTH = back, EAST = right, WEST = left
+	paramslist = params2list(params)
+	var/xOffset = text2num(paramslist["icon-x"]) - 16
+	var/yOffset = text2num(paramslist["icon-y"]) - 16
+
+	switch(H.dir)					//This makes the attackSide be the side that currently faces the 'camera'
+		if(NORTH)
+			attackedSide = SOUTH
+		if(SOUTH)
+			attackedSide = NORTH
+		if(EAST)
+			attackedSide = EAST
+		if(WEST)
+			attackedSide = WEST
+
+	var/icon/new_stamp_mark = icon('icons/effects/stamp_marks.dmi', "stamp[rand(1,3)]_[attackedSide]")
+	new_stamp_mark.Shift(EAST, xOffset)
+	new_stamp_mark.Shift(NORTH, yOffset)
+
+	var/mutable_appearance/targetBaseIcon = mutable_appearance()			//we will use this so the stamps only appear on the human's body and not
+	targetBaseIcon.dir = H.dir												//on their backpack or items they hold in their hand
+	targetBaseIcon.overlays += H.overlays_standing[BODY_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[LIMBS_LAYER]				//Currently this is here and not in update_icons because my only known
+	targetBaseIcon.overlays += H.overlays_standing[TAIL_LAYER]
+	targetBaseIcon.overlays += H.overlays_standing[UNDERWEAR_LAYER]			//method of turning an image into an icon destroys its directionality
+	targetBaseIcon.overlays += H.overlays_standing[HAIR_LAYER]				//	->and it needs to be an icon for the blending step
+	targetBaseIcon.overlays += H.overlays_standing[UNIFORM_LAYER]			//some species' bodies dont take up all the space their uniforms do
+	new_stamp_mark.Blend(getFlatIcon(targetBaseIcon), BLEND_MULTIPLY)		//cut out any parts of the stamp mark that aren't on base human
+
+	var/image/stamp_image = image(new_stamp_mark)
+	stamp_image.text = stamp_examine_list[icon_state]
```

Comment:
On the second thought you don't even need to keep them coupled?
Just have a list of stamps (descriptions), and a (single!) image of all(!) the stamps on the person (you don't even need to separate different stamps into different images).

Reply:
This is all real good, I will attempt to recode it this evening. I originally used .text to avoid lists of lists, but what you have suggested here is a great alternative.